theseus/utils/examples/bundle_adjustment/util.py

```diff
@@ -0,0 +1,57 @@
+import math
+from typing import Tuple
+
+import torch
+
+
+def soft_loss_cauchy(
+    x: torch.Tensor, radius: torch.Tensor
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    ratio = (x + radius) / radius
+    val = torch.log(ratio) * radius
+    der = 1.0 / ratio
+    return val, der
+
+
+def soft_loss_huber_like(
```

Comment:
I suspect many of these will be useful for @ddetone in his homography example too so we could move them up to main utils?